dcc-import-gene/src/main/java/org/icgc/dcc/imports/gene/thread/ASNInputReader.java

```diff
@@ -15,25 +15,36 @@
  * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package org.icgc.dcc.imports.gene.util;
+package org.icgc.dcc.imports.gene.thread;
 
-import lombok.RequiredArgsConstructor;
+import java.io.InputStream;
+import java.io.OutputStream;
 
-import org.icgc.dcc.imports.gene.core.GeneFilter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
 
-import com.fasterxml.jackson.databind.JsonNode;
+@Slf4j
+public class ASNInputReader implements Runnable {
 
-@RequiredArgsConstructor
-public class AllGeneFilter implements GeneFilter {
+  private InputStream in;
+  private OutputStream out;
 
-  @Override
-  public boolean filter(JsonNode gene) {
-    // Include all
-    return true;
+  public ASNInputReader(InputStream in, OutputStream out) {
+    this.in = in;
+    this.out = out;
   }
 
-  public static AllGeneFilter all() {
-    return new AllGeneFilter();
+  @Override
+  @SneakyThrows
+  public void run() {
+    int n;
+    byte[] buffer = new byte[16384];
```

Comment:
Seems pretty low level. Can't we Just use http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/io/ByteStreams.html#copy(java.io.InputStream,%20java.io.OutputStream)?